CondTools/RunInfo/plugins/LHCInfoPerFillPopConAnalyzer.cc

```diff
@@ -367,9 +363,14 @@ class LHCInfoPerFillPopConSourceHandler : public popcon::PopConSourceHandler<LHC
       size_t niovs = theLHCInfoPerFillImpl::transferPayloads(m_tmpBuffer, m_iovs, m_prevPayload);
       edm::LogInfo(m_name) << "Added " << niovs << " iovs within the Fill time";
       m_tmpBuffer.clear();
-      // iovAdded = true;
-      if (m_prevPayload->fillNumber() and !ongoingFill)
-        addEmptyPayload(endFillTime);
+      if (m_prevPayload->fillNumber() and !ongoingFill) {
+        if (m_endFillMode)
+          addEmptyPayload(endFillTime);
+        else {
+          cond::Time_t myLastLSiov = cond::lhcInfoHelper::getFillLastLumiIOV(oms, lhcFill);
+          addEmptyPayload(myLastLSiov);
```

Comment:
why not just?:
```
          addEmptyPayload(cond::lhcInfoHelper::getFillLastLumiIOV(oms, lhcFill));
```
But if you think it's better in 2 lines I would change `myLastLSiov` to `currentFillLastLsIov`